dali/operators/reader/fits_reader_op.cc

```diff
@@ -0,0 +1,116 @@
+
+#include <string>
+
+#include "dali/core/backend_tags.h"
+#include "dali/core/static_switch.h"
+#include "dali/kernels/slice/slice_cpu.h"
+#include "dali/kernels/slice/slice_flip_normalize_permute_pad_cpu.h"
+#include "dali/kernels/transpose/transpose.h"
+#include "dali/operators/reader/fits_reader_op.h"
+#include "dali/pipeline/data/backend.h"
+
+namespace dali {
+namespace detail {
+inline int FitsReaderOutputFn(const OpSpec &spec) {
+  // there is default value provided, so no checking if arg exists is needed
+  return spec.GetRepeatedArgument<int>("hdu_indices").size();
+}
+
+}  // namespace detail
+
+DALI_REGISTER_OPERATOR(experimental__readers__Fits, FitsReaderCPU, CPU);
+
+DALI_SCHEMA(experimental__readers__Fits)
+    .DocStr(R"(Reads Fits image HDUs from a directory.
+
+This operator can be used in the following modes:
+
+1. Read all files from a directory indicated by ``file_root`` that match given ``file_filter``.
+2. Read file names from a text file indicated in ``file_list`` argument.
+3. Read files listed in ``files`` argument.
+)")
+    .NumInput(0)
+    .OutputFn(detail::FitsReaderOutputFn)
+    .AddOptionalArg<string>("file_root",
+                            R"(Path to a directory that contains the data files.
+
+If not using ``file_list`` or ``files``, this directory is traversed to discover the files.
+``file_root`` is required in this mode of operation.)",
+                            nullptr)
+    .AddOptionalArg(
+        "file_filter",
+        R"(If a value is specified, the string is interpreted as glob string to filter the
+list of files in the sub-directories of the ``file_root``.
+
+This argument is ignored when file paths are taken from ``file_list`` or ``files``.)",
+        "*.fits")
+    .AddOptionalArg<string>("file_list",
+                            R"(Path to a text file that contains filenames (one per line)
+where the filenames are relative to the location of that file or to ``file_root``, if specified.
+
+This argument is mutually exclusive with ``files``.)",
+                            nullptr)
+    .AddOptionalArg("shuffle_after_epoch",
+                    R"(If set to True, the reader shuffles the entire dataset after each epoch.
+
+``stick_to_shard`` and ``random_shuffle`` cannot be used when this argument is set to True.)",
+                    false)
+    .AddOptionalArg<vector<string>>("files", R"(A list of file paths to read the data from.
+
+If ``file_root`` is provided, the paths are treated as being relative to it.
+
+This argument is mutually exclusive with ``file_list``.)",
+                                    nullptr)
+    .AddOptionalArg("hdu_indices",
+                    R"(HDU indexes to read. If not provided, first HDU after primary 
+                                 R"(HDU indexes to read. If not provided, first HDU after primary 
+                    R"(HDU indexes to read. If not provided, first HDU after primary 
+  will be yielded (i.e. will default to  hdu_indices=[2]).)",
```

Comment:
```suggestion
                    R"(HDU indexes to read. If not provided, first HDU after primary 
  will be yielded.)",
```
I think that the default is generated automatically in the docs. Looks like the line got duplicated.
You can maybe mention that the indexing starts from 1.